lib/facter/resolvers/augeas_resolver.rb

```diff
@@ -14,10 +14,28 @@ def post_resolve(fact_name)
         end
 
         def read_augeas_version(fact_name)
-          output = Facter::Core::Execution.execute('augparse --version 2>&1', logger: log)
-          @fact_list[:augeas_version] = Regexp.last_match(1) if output =~ /^augparse (\d+\.\d+\.\d+)/
+          @fact_list[:augeas_version] = read_augeas_from_cli
+          @fact_list[:augeas_version] ||= read_augeas_from_gem
+
           @fact_list[fact_name]
         end
+
+        def read_augeas_from_cli
+          output = Facter::Core::Execution.execute('augparse --version 2>&1', logger: log)
+          Regexp.last_match(1) if output =~ /^augparse (\d+\.\d+\.\d+)/
+        end
+
+        def read_augeas_from_gem
+          require 'augeas'
+          aug = ::Augeas.open('/', nil, ::Augeas::NO_MODL_AUTOLOAD)
+          ver = aug.get('/augeas/version')
+          aug.close
+          ver
```

Comment:
you should use open with a block so the close is always ensured.
Also there is another gem that provides `Augeas` interface: https://rubygems.org/gems/augeas/versions/0.6.4, in which the open methods is private and it was replaced with `.create`.
This will not work on systems where users install facter as a gem and have augeas > 0.5.

Reply:
Used `Augeas.create` when `Augeas` responds to `create` (version > 0.5.0) and `Augeas.open` when `Auges` responds to `open` (version <= 0.5.0)